integrationtests/src/test/java/io/kroxylicious/proxy/multitenant/MultiTenantIT.java

```diff
@@ -435,4 +444,33 @@ private static <K, V> Predicate<ConsumerRecord<K, V>> matchesRecord(final String
             return Objects.equals(rec.topic(), expectedTopic) && Objects.equals(rec.key(), expectedKey) && Objects.equals(rec.value(), expectedValue);
         };
     }
+
+    private static class PartitionAssignmentAwaitingRebalanceListener<K, V> implements ConsumerRebalanceListener {
+        private final CompletableFuture<Object> future = new CompletableFuture<>();
```

Comment:
Feels like a [CountdownLatch](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/CountDownLatch.html) would be better suited to the role of awaiting the rebalance. Or maybe an [AtomicBoolean](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/atomic/AtomicBoolean.html) which has the advantage of playing better with Awaitility if we choose that door.